### PR TITLE
Itembox Implementation

### PR DESCRIPTION
**Pull request by FastestMolasses**
### Description

Adds the ItemBox component, which is used for the auto complete feature.

### Checklist

- [x] I read and understood the [contributing guide](https://github.com/CodeEditApp/CodeEdit/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/CodeEditApp/CodeEdit/blob/main/CODE_OF_CONDUCT.md)
- [x] The issues this PR addresses are related to each other
- [x] My changes generate no new warnings
- [x] My code builds and runs on my machine
- [x] My changes are all related to the related issue above
- [x] I documented my code

### CODE REVIEW

File: Sources/CodeEditTextView/ItemBox/ItemBoxWindowController.swift [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+class NoSlotScroller: NSScroller {
+    override class var isCompatibleWithOverlayScrollers: Bool { true }
+
+    override func drawKnobSlot(in slotRect: NSRect, highlight flag: Bool) {
+        // Don't draw the knob slot (the background track behind the knob)
+    }
+}
+
+public protocol ItemBoxDelegate: AnyObject {
```
Comment (thecoolwinter) [inline Sources/CodeEditTextView/ItemBox/ItemBoxWindowController.swift:238]: This should also be in it's own file

File: Sources/CodeEditTextView/ItemBox/ItemBoxWindowController.swift [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+import AppKit
+import LanguageServerProtocol
+
+/// Represents an item that can be displayed in the ItemBox
+public protocol ItemBoxEntry {
+    var view: NSView { get }
+}
+
+/// Padding at top and bottom of the window
+let WINDOW_PADDING: CGFloat = 5
```
Comment (tom-ludwig) [inline Sources/CodeEditTextView/ItemBox/ItemBoxWindowController.swift:17]: We're a bit inconsistent with this. Here you declared the constant with capital letters and in the extension you have constants declared normally, within the class and in lowercase.

**Review by thecoolwinter — CHANGES_REQUESTED** [review]
I agree with @matthijseikelenboom this should be in the CESE repo.

We discussed briefly what it should be called, and I think you're right that it could be used for other things in the future but calling it `CompletionWindow` would be fine in that package. It can be extended later but we can avoid that complexity for now.

Also if you move it, you should be able to add the keybind you have in CE right now to the existing keybind handler in CESE. That might fix the issue you mentioned with not being able to close tabs. You'd then also have access to the theme colors for styling this window.

**Review by tom-ludwig — CHANGES_REQUESTED** [review]
Amazing! I was wondering if it is necessary to add this functionality here or rather in CESE, what do you think?

**Review by matthijseikelenboom — APPROVED** [review]
Like I said In Discord, I'd suggest changing the name "ItemBox" to something more clearer. Other than the comments left by the other guys, I think it looks good

**Pull request closed without merging**